report.go

```diff
@@ -86,16 +83,21 @@ func report(files []string, typ, output string, every time.Duration) error {
 	case "json":
 		var m vegeta.Metrics
 		rep, report = vegeta.NewJSONReporter(&m), &m
-	case "hist":
-		if len(typ) < 6 {
-			return fmt.Errorf("bad buckets: '%s'", typ[4:])
-		}
-		var hist vegeta.Histogram
-		if err := hist.Buckets.UnmarshalText([]byte(typ[4:])); err != nil {
-			return err
-		}
-		rep, report = vegeta.NewHistogramReporter(&hist), &hist
+	case "hdrplot":
+		var m vegeta.Metrics
+		rep, report = vegeta.NewHDRHistogramPlotReporter(&m), &m
 	default:
+		switch {
+		case strings.HasPrefix(typ, "hist"):
+			if len(typ) < 6 {
+				return fmt.Errorf("bad buckets: '%s'", typ[4:])
+			}
+			var hist vegeta.Histogram
+			if err := hist.Buckets.UnmarshalText([]byte(typ[4:])); err != nil {
+				return err
+			}
+			rep, report = vegeta.NewHistogramReporter(&hist), &hist
+		}
 		return fmt.Errorf("unknown report type: %q", typ)
```

Comment:
With this change you'll always return the error for -typ=hist, as you have no return after the `rep, report =` line.